ngraph_bridge/ngraph_builder.cc

```diff
@@ -4933,6 +4933,26 @@ static Status TranslateUnpackOp(const Node* op,
   return Status::OK();
 }
 
+static Status TranslateXdivyOp(
+    const Node* op, const std::vector<const Tensor*>& static_input_map,
+    Builder::OpMap& ng_op_map) {
+  
+  shared_ptr<ng::Node> ng_input1, ng_input2, ng_input3;
+  TF_RETURN_IF_ERROR( GetInputNodes(ng_op_map, op, &ng_input1, &ng_input2));
+  std::tie(ng_input1, ng_input2) = Builder::PerformNgBroadcast(op->name(),ng_input1, ng_input2);
+  ng::Shape input1_shape = ng_input1->get_shape();
+  std::vector<std::string> const_values(ng::shape_size(input1_shape), "0");
+  auto const_zero= ConstructNgNode<ng::op::Constant>( op->name(), ng_input1->get_element_type(), input1_shape, const_values); 
+  auto ng_is_zero = ConstructNgNode<ng::op::Equal>(op->name(),ng_input1, const_zero);
+  auto ng_x_over_y = ConstructNgNode<ng::op::Divide>(op->name(), ng_input1, ng_input2);
+  auto ng_xdivy = ConstructNgNode<ng::op::Select>(op->name(),ng_is_zero, ng_input1, ng_x_over_y);
+  if(ng_xdivy!= ng_input1 && ng_xdivy!= ng_input2){
```

Comment:
We do not need this `if`
We need this if only in functions that accept ng nodes and return new ng nodes (and the function might be creating a small ng subgraph). Eg ngraph builders